lib/instrumentation/modules/kafkajs.js

```diff
@@ -356,13 +367,15 @@ module.exports = function (mod, agent, { version, enabled }) {
    */
   function wrapProducerSendBatch(origSendBatch) {
     return async function (batch) {
-      // TODO: discuss this. if all topics are ignored should the agent avoid
-      // creating a span?
       let span;
-      const topics =
-        batch.topicMessages && batch.topicMessages.map((tm) => tm.topic);
-      const shouldIgnoreBatch =
-        topics && topics.every((t) => shouldIgnoreTopic(t, config));
+      let shouldIgnoreBatch = true;
+      const messages = batch.topicMessages || [];
+      for (const msg of messages) {
+        if (!shouldIgnoreTopic(msg.topic, config)) {
+          shouldIgnoreBatch = false;
+          break;
+        }
+      }
```

Comment:
nit: This nit might be premature optimization. If the number of messages is very large (say 1000), then we'll be running the possibly-relatively-slow `shouldIgnoreTopic` for every message, when those many messages are likely all from a small set of topics. Perhaps this could get the set of topics and only `shouldIgnoreTopic(topic)` once for each topic.

Reply:
yes, the previous code was doing the same and also allocating a new array of strings and using a callback function as an argument for `.every` method. So this version tackles only the memory allocation.

What I've missed is a topic could be the same for many messages so we could use a `Set` and iterate over it. I'm going to add this change